packages/experimental/Divider/src/Divider.tsx

```diff
@@ -0,0 +1,81 @@
+/** @jsx withSlots */
+/** @jsxFrag */
+import React from 'react';
+import { View } from 'react-native';
+import type { ViewProps } from 'react-native';
+import { dividerName } from './Divider.types';
+import type { DividerProps, DividerTokens } from './Divider.types';
+import { withSlots, compressible, useSlot, useFluentTheme, patchTokens, mergeStyles } from '@fluentui-react-native/framework';
+import type { UseTokens } from '@fluentui-react-native/framework';
+import { TextV1 as Text } from '@fluentui-react-native/text';
+import type { TextProps } from '@fluentui-react-native/text';
+import { IconV1 as Icon } from '@fluentui-react-native/icon';
+import type { IconPropsV1 as IconProps } from '@fluentui-react-native/icon';
+import { useDividerTokens } from './DividerTokens';
+import { globalTokens } from '@fluentui-react-native/theme-tokens';
+import { colorsFromPropsAndTheme, useDividerSlotProps } from './Divider.styling';
+
+export const Divider = compressible<DividerProps, DividerTokens>((props: DividerProps, useTokens: UseTokens<DividerTokens>) => {
+  const { alignContent = 'center', appearance = 'default', color, icon, insetSize = 0, vertical = false } = props;
+
+  const theme = useFluentTheme();
+  let [tokens, cache] = useTokens(theme);
+
+  // We want to patch our initial tokens from useTokens() with the token values below because they are dependent on the above props.
+  // With compressible, we can also access our tokens at the final rendering stage, which we will need to correctly set our divider height.
+  [tokens, cache] = patchTokens(tokens, cache, {
+    flexAfter: alignContent === 'end' ? 0 : 1,
+    flexBefore: alignContent === 'start' ? 0 : 1,
+    minHeight: vertical ? globalTokens.size240 : 0,
+    ...colorsFromPropsAndTheme({ appearance, color }, theme),
+  });
+
+  // get slot props from these tokens
+  const { rootProps, beforeLineProps, afterLineProps, wrapperProps, textProps, iconProps } = useDividerSlotProps(
+    { icon, insetSize, vertical },
```

Comment:
this would invalidate the caching done in useDividerSlotProps. We should memoize this object creation.